sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -602,6 +604,9 @@ private static void initializeRegistry() {
       Class<T> iface, Set<Class<? extends PipelineOptions>> validatedPipelineOptionsInterfaces) {
     checkArgument(iface.isInterface(), "Only interface types are supported.");
 
+    // Validate that all interfaces are inherited from PipelineOptions.
```

Comment:
`Validate that all interfaces are inherited from PipelineOptions.` -> `Validate that every inherited interface must extend PipelineOptions except for PipelineOptions itself.`